File: 백준/Silver/2606. 바이러스/바이러스.java
import java.io.*;
import java.util.StringTokenizer;

public class Main {
    static boolean visited[];
    static int virus[][];
    static int cnt = 0; //  전파 될 컴퓨터 수
    static int com;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        com = Integer.parseInt(br.readLine());  //  컴퓨터 수
        int net = Integer.parseInt(br.readLine());  //  컴퓨터 쌍의 수

        visited = new boolean[com + 1];  //  감염 여부
        virus = new int[com + 1][com + 1];  //  바이러스에 걸린 컴퓨터
        int startcom = 1;   //  1번 컴퓨터 숙주

        for (int i = 0; i < net; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());   //  쌍 1
            int b = Integer.parseInt(st.nextToken());   //  쌍 2

            virus[a][b] = virus[b][a] = 1;
        }

        sb.append(dfs(startcom));
        System.out.println(sb.toString());
    }

    static int dfs(int startcom) {
        visited[startcom] = true;  //  숙주는 감염되었다.(초기값)

        for (int i = 1; i <= com; i++) {//  나머지 감염여부 확인
            if (virus[startcom][i] == 1 && !visited[i]) {   //  감염되지 않았으면서 숙주와 연결된다면
                cnt++;  //  감염 컴퓨터 수 증가
                dfs(i); //  다른 감염된 애들로 다시 찾기
            }
        }
        return cnt;
    }
}

File: 백준/Silver/2606. 바이러스/바이러스.py
import sys

c = int(sys.stdin.readline())
n = int(sys.stdin.readline())

graph = [[0] * (c + 1) for _ in range(c + 1)]
isVisited = [False] * (c + 1)

for i in range(n):
    a,b = map(int, sys.stdin.readline().split())
    graph[a][b] = 1
    graph[b][a] = 1

cnt = 0

def dfs(v):
    global cnt
    isVisited[v] = True
    for i in range(1, c + 1):
        if not isVisited[i] and graph[v][i] == 1:
            cnt += 1
            dfs(i)

dfs(1)
print(cnt)
